### ISSUE TITLE
System libraries automatically added to ScaleUnit extension directory cause error at runtime in release/9.39

### ISSUE THREAD

**Issue by rphillips-newestech** [microsoft/Dynamics365Commerce.ScaleUnit#23]
I have a customization that was working fine with release/9.38. I am attempting to update to release/9.39. Updated to release/9.39, my customization builds and installs just fine. I can even see my custom endpoints listed in the RS' metadata view. However, whenever the POS (store commerce) attempts to interact with any of the custom RS endpoints, the request fails with the error at the end of this post.

I looked at the extensions installation directory and a bunch of Microsoft and system libraries have been added in addition to my custom libraries. I don't believe that this is normal, as when I install the [example ScaleUnit extension](https://github.com/microsoft/Dynamics365Commerce.ScaleUnit/tree/release/9.39/src/ScaleUnitSample), none of these libraries exist. _Also, if I manually remove all of the files except for my customization libraries and restart the server then interaction with the RS endpoints from the POS is successful._

> My question is: **what is causing my custom ScaleUnit installer to include these additional libraries? How can I fix this without manual intervention?**

Except for Newtonsoft.Json, my various project files only reference libraries and versions specified in [repo.props](https://github.com/microsoft/Dynamics365Commerce.ScaleUnit/blob/release/9.39/repo.props). As mentioned, the only other library referenced by my code is Newtonsoft.Json which is referencing version 13.0.1.

`<PackageReference Include="Newtonsoft.Json" Version="13.0.1" />`

I chose this version of Newtonsoft, because this is the version that the 9.39.22263.8 version of the base sealed scale unit installs, specifically 13.0.1.25517. This is the latest base sealed scale unit installer on LCS as of 9/20. The Newtonsoft library might be a rabbit trail, but it is the only significant difference I am seeing in my customization's project files v. the example ScaleUnit extension.

-----
### Microsoft and system libraries added to extension directory
```
/ChannelDatabase
/POS
(... my custom libraries ...)
Extension.config
Microsoft.Dynamics.Retail.Diagnostics.dll
Microsoft.Dynamics.Retail.PaymentSDK.Portable.dll
Microsoft.Extensions.DependencyInjection.Abstractions.dll
Microsoft.Extensions.DependencyInjection.dll
Microsoft.Extensions.Logging.Abstractions.dll
Microsoft.Extensions.Logging.Debug.dll
Microsoft.Extensions.Logging.dll
Microsoft.Extensions.Options.dll
Microsoft.Extensions.Primitives.dll
Newtonsoft.Json.dll
System.Collections.Concurrent.dll
System.ComponentModel.dll
System.Composition.AttributedModel.dll
System.Composition.Convention.dll
System.Composition.Hosting.dll
System.Composition.Runtime.dll
System.Composition.TypedParts.dll
System.Diagnostics.DiagnosticSource.dll
System.IO.FileSystem.Primitives.dll
System.Linq.dll
System.Private.DataContractSerialization.dll
System.Reflection.Emit.dll
System.Reflection.Emit.ILGeneration.dll
System.Reflection.Emit.Lightweight.dll
System.Reflection.TypeExtensions.dll
System.Runtime.CompilerServices.Unsafe.dll
System.Runtime.Numerics.dll
System.Runtime.Serialization.Json.dll
System.Runtime.Serialization.Primitives.dll
System.Runtime.Serialization.Xml.dll
System.Security.Cryptography.OpenSsl.dll
System.Security.Cryptography.Primitives.dll
System.Text.RegularExpressions.dll
System.Threading.dll
System.Threading.Tasks.Extensions.dll
System.Xml.ReaderWriter.dll
System.Xml.XDocument.dll
System.Xml.XmlDocument.dll
System.Xml.XmlSerializer.dll
```
-----
```
### Error
`An error occurred during the Retail Server Request. RequestUri: https://(...retail server address...)/RetailServer/Commerce/(...method name...)?api-version=7.3. RequestId: b4fc2e1e-d7a2-b397-e490-1ceb80013033. Exception: System.TypeLoadException: Could not load type 'Microsoft.Extensions.DependencyInjection.ActivatorUtilitiesConstructorAttribute' from assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=1.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.
   at System.ModuleHandle.ResolveType(QCallModule module, Int32 typeToken, IntPtr* typeInstArgs, Int32 typeInstCount, IntPtr* methodInstArgs, Int32 methodInstCount, ObjectHandleOnStack type)
   at System.ModuleHandle.ResolveTypeHandle(Int32 typeToken, RuntimeTypeHandle[] typeInstantiationContext, RuntimeTypeHandle[] methodInstantiationContext)
   at System.Reflection.RuntimeModule.ResolveType(Int32 metadataToken, Type[] genericTypeArguments, Type[] genericMethodArguments)
   at System.Reflection.CustomAttribute.FilterCustomAttributeRecord(MetadataToken caCtorToken, MetadataImport& scope, RuntimeModule decoratedModule, MetadataToken decoratedToken, RuntimeType attributeFilterType, Boolean mustBeInheritable, ListBuilder1& derivedAttributes, RuntimeType& attributeType, IRuntimeMethodInfo& ctorWithParameters, Boolean& isVarArg)
   at System.Reflection.CustomAttribute.IsCustomAttributeDefined(RuntimeModule decoratedModule, Int32 decoratedMetadataToken, RuntimeType attributeFilterType, Int32 attributeCtorToken, Boolean mustBeInheritable)
   at System.Reflection.RuntimeConstructorInfo.IsDefined(Type attributeType, Boolean inherit)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.TryFindPreferredConstructor(Type instanceType, Type[] argumentTypes, ConstructorInfo& matchingConstructor, Nullable1[]& parameterMap)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.FindApplicableConstructor(Type instanceType, Type[] argumentTypes, ConstructorInfo& matchingConstructor, Nullable1[]& matchingParameterMap)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateFactory(Type instanceType, Type[] argumentTypes)
   at Microsoft.AspNetCore.Mvc.Controllers.ControllerActivatorProvider.CreateActivator(ControllerActionDescriptor descriptor)
   at Microsoft.AspNetCore.Mvc.Controllers.ControllerFactoryProvider.CreateControllerFactory(ControllerActionDescriptor descriptor)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvokerCache.GetCachedResult(ControllerContext controllerContext)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvokerProvider.OnProvidersExecuting(ActionInvokerProviderContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionInvokerFactory.CreateInvoker(ActionContext actionContext)
   at Microsoft.AspNetCore.Mvc.Routing.ActionEndpointFactory.<>c__DisplayClass9_0.<CreateRequestDelegate>b__0(HttpContext context)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext)
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNet.OData.Batch.ODataBatchMiddleware.Invoke(HttpContext context)
   at Microsoft.Dynamics.Retail.RetailServerLibrary.Middlewares.Redirection.RedirectionMiddleware.Invoke(HttpContext context)
   at IdentityServer4.Hosting.IdentityServerMiddleware.Invoke(HttpContext context, IEndpointRouter router, IUserSession session, IEventService events, IBackChannelLogoutService backChannelLogoutService)
   at Microsoft.Dynamics.Retail.RetailServerLibrary.Middlewares.Authentication.ExternalIdentityAuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.Dynamics.Retail.RetailServerLibrary.Middlewares.Authentication.OperatingUnitAuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.Dynamics.Retail.RetailServerLibrary.Middlewares.Authentication.DeviceTokenAuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at IdentityServer4.Hosting.BaseUrlMiddleware.Invoke(HttpContext context)
   at Microsoft.Dynamics.Retail.RetailServerLibrary.Middlewares.Instrumentation.InstrumentationMiddleware.Invoke(HttpContext context). UnscrubbedException: .
```

**Comment by lukedgr** [comment]
Hi @rphillips-newestech , could you please build your solution using msbuild with -verbosity:diag and upload the log file here?

**Comment by rphillips-newestech** [comment]
Hey @lukedgr, Thanks for the reply. I will get you that log early next week. I had hoped to explore this a little more myself and maybe provide a simplified example, but have been slammed with other work.

**Comment by rphillips-newestech** [comment]
@lukedgr, I am no longer encountering this issue. It appears that it has been resolved with time. Unless you would still like me to provide some additional information, feel free to close the case.

I set up a new .29 environment and reattempted updating my customization to target release/9.39. This time there were no issues interacting with the scale unit and the customization performed as expected.
